pyanaconda/modules/storage/devicetree/fsset.py

```diff
@@ -591,6 +591,15 @@ def mount_filesystems(self, root_path="", read_only=None, skip_root=False):
             if read_only:
                 options = "%s,%s" % (options, read_only)
 
+            # Create /tmp with the right permissions (rhbz#1937626).
+            # It needs to be created right before we mount anything
+            if device.format.mountpoint == "/tmp":
+                path = os.path.join(root_path, "tmp")
+
+                if not os.path.exists(path):
+                    os.makedirs(path)
+                    os.chmod(path, 0o1777)
```

Comment:
I think you can call makedirs with the mode already. You actually check for the default in the test...

Reply:
Unfortunately, `makedirs` doesn't enforce the mode (see [the documentation of mkdir](https://docs.python.org/3/library/os.html#os.mkdir)), so we have to call `chmod`. I have changed the test, because it didn't check the right call.